### PR TITLE
#666: Refactored the class RqMtBase to reduce the coupling

### PR DESCRIPTION
**Pull request by essobedo**
Fix for #666 

The goal of this PR is to refactor the class RqMtBase in order to reduce the overall coupling pf this class

### CODE REVIEW

**Comment by davvd** [comment]
@essobedo Thanks, I will find someone to review this PR soon

**Comment by davvd** [comment]
@dmzaytsev review this one,please

File: src/test/java/org/takes/rq/RqTempTest.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+     * RqTemp can delete the underlying temporary file.
+     * @throws IOException if some problem occurs.
+     */
+    @Test
+    public void deletesTempFile() throws IOException {
+        final File file = File.createTempFile(
+            RqTempTest.class.getName(),
+            ".tmp"
+        );
+        final BufferedWriter out = new BufferedWriter(new FileWriter(file));
```
Comment (dmzaytsev) [inline src/test/java/org/takes/rq/RqTempTest.java]: @essobedo I think it can be implemented shortly as Takes moved to java 7
```
Files.write(
    file.toPath(), Arrays.asList(""Temp file deletion test""), StandardCharsets.UTF_8
);
```

**Comment by dmzaytsev** [comment]
@essobedo please see my comment

**Comment by essobedo** [comment]
@dmzaytsev done, thx for the feedback

**Comment by dmzaytsev** [comment]
@essobedo thanks

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by essobedo** [comment]
@yegor256 please check

**Comment by essobedo** [comment]
@yegor256 ping

**Comment by essobedo** [comment]
@yegor256 Please confirm

**Comment by essobedo** [comment]
@yegor256 please

File: src/main/java/org/takes/rq/multipart/RqMtBase.java [hunk 2]
```diff
@@ -140,15 +129,15 @@ public RqMtBase(final Request req) throws IOException {
         if (values == null) {
             iter = new VerboseIterable<>(
                 Collections.<Request>emptyList(),
-                new Sprintf(
+                String.format(
```
Comment (yegor256) [inline src/main/java/org/takes/rq/multipart/RqMtBase.java]: @essobedo what is the reason for this change?

File: src/main/java/org/takes/rq/multipart/RqMtBase.java [hunk 3]
```diff
@@ -140,15 +129,15 @@ public RqMtBase(final Request req) throws IOException {
...
                     "there are no parts by name \"%s\" among %d others: %s",
                     name, this.map.size(), this.map.keySet()
                 )
             );
         } else {
             iter = new VerboseIterable<>(
                 values,
-                new Sprintf(
+                String.format(
```
Comment (yegor256) [inline src/main/java/org/takes/rq/multipart/RqMtBase.java]: @essobedo same here. why?

**Comment by yegor256** [comment]
@essobedo some questions from me above

File: src/main/java/org/takes/rq/multipart/RqMtBase.java (hunk 2 above)
Comment (essobedo) [inline src/main/java/org/takes/rq/multipart/RqMtBase.java]: @yegor256 I needed to reduce the coupling, and as I tested the result of both and got the same result, I believed that I could replace it. Why is it not correct? What did I miss? What does Sprintf that String.format doesn't do?

**Comment by essobedo** [comment]
@yegor256 I provided the answer, please check it

**Comment by essobedo** [comment]
@yegor256 please check my answers

**Comment by essobedo** [comment]
@yegor256 please check my answer

**Comment by essobedo** [comment]
@yegor256 ping

File: src/main/java/org/takes/rq/RqLive.java [hunk 4]
```diff
@@ -45,7 +45,7 @@
  * @since 0.1
  */
 @EqualsAndHashCode(callSuper = true)
-public final class RqLive extends RqWrap {
+public class RqLive extends RqWrap {
```
Comment (yegor256) [inline src/main/java/org/takes/rq/RqLive.java]: @essobedo check this: http://www.yegor256.com/2014/11/20/seven-virtues-of-good-object.html#7-his-class-is-either-final-or-abstract

File: src/main/java/org/takes/rq/multipart/RqMtBase.java (hunk 2 above)
Comment (yegor256) [inline src/main/java/org/takes/rq/multipart/RqMtBase.java]: @essobedo `String.format()` will print the string right here, while `Sprintf` will print only when it's needed. in most cases - never.

**Comment by yegor256** [comment]
@essobedo please excuse me for the delay. one more comment above

**Comment by essobedo** [comment]
@yegor256 fixed, please check again

File: src/main/java/org/takes/rq/RqTemp.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+ * @see org.takes.rq.RqLive
+ * @see org.takes.rq.TempInputStream
+ */
+@EqualsAndHashCode(callSuper = true)
+public final class RqTemp extends RqWrap {
+
+    /**
+     * Creates a {@code RqTemp} with the specified temporary file.
+     * @param file The temporary that will be automatically deleted when the
+     *  body of the request will be closed.
```
Comment (yegor256) [inline src/main/java/org/takes/rq/RqTemp.java]: @essobedo I don't see where this file is actually deleted... do I miss something?

Comment (yegor256) [inline src/main/java/org/takes/rq/RqTemp.java]: @essobedo ignore me

File: src/main/java/org/takes/rq/RqTemp.java [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+/**
+ * Live request with a temporary file as body.
+ * @author Nicolas Filotto (nicolas.filotto@gmail.com)
+ * @version $Id$
+ * @since 0.33
+ * @see org.takes.rq.RqLive
+ * @see org.takes.rq.TempInputStream
+ */
+@EqualsAndHashCode(callSuper = true)
+public final class RqTemp extends RqWrap {
```
Comment (yegor256) [inline src/main/java/org/takes/rq/RqTemp.java]: @essobedo you're making this class public, but I can't imagine why anyone would need it... we need a better JavaDoc explanation of what it's for... with an example or something. now it's absolutely not clear what is it all about

**Comment by yegor256** [comment]
@essobedo see my concern above

File: src/main/java/org/takes/rq/RqTemp.java (hunk 6 above)
Comment (essobedo) [inline src/main/java/org/takes/rq/RqTemp.java]: @yegor256 I agree it doesn't need to be public. OK I will improve the javadoc

**Comment by essobedo** [comment]
@yegor256 RqTemp is no more public and has been moved into the multipart package, please check again

**Comment by yegor256** [comment]
@rultor try to merge

**Comment by davvd** [comment]
@elenavolokhova please, check this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by davvd** [comment]
@rultor please deploy

**Comment by elenavolokhova** [comment]
@dmzaytsev 
According to our [quality rules](http://www.teamed.io/qa.html):
 > The code reviewer found at least three problems in the code.
 
Only one issue was found during review.
Please confirm that you will try to find at least three problems next time.

**Comment by dmzaytsev** [comment]
@elenavolokhova I confirm

**Comment by elenavolokhova** [comment]
@davvd Quality is acceptable here.

**Comment by davvd** [comment]
> @davvd Quality is acceptable here.

@elenavolokhova thank you, let's try to make it "good" next time

**Comment by davvd** [comment]
@dmzaytsev added 10 mins to @elenavolokhova, for QA review, payment code is `85775163`

thanks, paid, **20 mins** to your account, payment ID is `85775198`, task took 241 hours and 33 mins to complete

you're getting extra minutes here (c=5)

+20 added to your rating, at the moment it is: [+1304](http://www.netbout.com/b/35166?open=rating)
